apps/site/components/withAvatarGroup.tsx

```diff
@@ -0,0 +1,36 @@
+'use client';
+
+import type { ComponentProps, FC } from 'react';
+
+import AvatarGroup from '@/components/Common/AvatarGroup';
+import { getAuthorWithId, getAuthorWithName } from '@/util/authorUtils';
+
+type WithAvatarGroupProps = Omit<
+  ComponentProps<typeof AvatarGroup>,
+  'avatars'
+> & {
+  usernames?: Array<string>;
+  names?: Array<string>;
+  clickable?: boolean;
+  container?: HTMLElement | null;
+};
+
+const WithAvatarGroup: FC<WithAvatarGroupProps> = ({
+  usernames,
+  names,
+  clickable = true,
+  ...props
+}) => (
+  <AvatarGroup
+    avatars={
+      usernames
+        ? getAuthorWithId(usernames, clickable)
```

Comment:
Do we need to nest?

Reply:
By adding type checking, I removed the final `[]`. I could simplify the functions and move them inside a single function, but wouldn’t it be better to keep them separate functions? 🤔

Reply:
I think it's fine as is. @ovflowd, if this is your last concern, than this should be good to land :-)

Reply:
I do usually prefer avoiding nested ?: chains as it makes the code harder to read

Reply:
@canerakdas you could use an `??` statement, and then in the utilities, if the input is null, return null

Reply:
There's no need to debate this too long, because things are going to change soon, as there was a plan to standardise the blog's frontmatter to use github users only.

Reply:
That doesn't mean we can avoid shipping tech debt from day 0 ;)

Reply:
I was able to update it a little late 🫡 sorry for the delay ❤️